plugin/storage/es/spanstore/writer.go

```diff
@@ -73,6 +75,13 @@ type Span struct {
 	StartTimeMillis uint64 `json:"startTimeMillis"`
 }
 
+func (s Service) hashCode() string {
+	h := fnv.New64a()
```

Comment:
can this be cached?

Reply:
@black-adder It could be but it will introduce issue with concurrency. Every simultaneous call to this function would potentially cause race condition and would render unpredictable hash results. I would need to introduce some kind of semaphore and mark this part as critical section. Do you think it is worth?

Reply:
@black-adder so my reasoning is like this.
1. You store cached hasher object somewhere (most likely in the writer.go) and use it whenever someone calls `hashCode()`
2. First process (1) calls `hashCode()`, fetches (creates if not cached) hasher object
3. (1) Pushes data: `OperationName` and `ServiceName` to the hasher
4. Second process (2) calls `hashCode()` which fetches the same (cached) hasher object
5. (2) pushes its `OperationName` and `ServiceName` to the same hasher
6. (1) calculates checksum by calling `Sum64()` on the same hasher (which now holds data from two possibly different instances of `Service`)
7. (2) does the calculations too and gets the same (wrong) hash as (1)

I did actually run into this issue while fixing the tests :-) 

Even if I introduce `hash.Reset()` just after `Sum64()` or before data is pushed it the hasher I still have no guarantees that other process is not doing some operation on the cached object. Putting Write semaphore would probably do the trick but I think it is a bit overkill in this case.

I hope this helps.